### PR TITLE
to review one is taking the list one is taking the type

### PR DESCRIPTION
**Pull request by ibrahim-mestadi**

### CODE REVIEW

**Review by oubaydos — COMMENTED** [review]
we are using randomGenerator instead of ThreadLocalRandom

**Review by oubaydos — CHANGES_REQUESTED** [review]
add javadocs and change visibility from "public" to "private"

**Comment by oubaydos** [comment]
otherwise nice work 😄

File: src/main/java/RandomGenerator.java [hunk 1]
```diff
@@ -201,10 +211,48 @@ private Map<Object, Object> getRandomMap(Type type) {
...
+
+
+    public <T> List<T> getRandomList(List<T> listTofill) throws NoSuchFieldException {
         // TODO @ibrahim
-        return null;
+        int randomSize = ThreadLocalRandom
+                .current()
+                .nextInt(MIN_COLLECTION_SIZE, MAX_COLLECTION_SIZE);
+
+        Field ListField = RandomGenerator.class.getDeclaredField("listTofill");
```
Comment (HamzaBenyazid) [inline src/main/java/RandomGenerator.java:224]: variable names should be lower camel case

File: src/main/java/RandomGenerator.java [hunk 2]
```diff
@@ -201,10 +211,48 @@ private Map<Object, Object> getRandomMap(Type type) {
...
+
+        return (List<T>) outputList;
+
     }
+    public <T> List<T> getRandomList(Type type) {
+        int randomSize = ThreadLocalRandom
+                .current()
+                .nextInt(MIN_COLLECTION_SIZE, MAX_COLLECTION_SIZE);
+        ParameterizedType parameterizedType = (ParameterizedType)type;
+        Class<?> inputListClass = (Class<?>) parameterizedType.getActualTypeArguments()[0];
```
Comment (HamzaBenyazid) [inline src/main/java/RandomGenerator.java:243]: no need to cast it to Class<?> , just use it as Type (see getRandomMap)

File: src/main/java/RandomGenerator.java [hunk 3]
```diff
@@ -201,10 +211,48 @@ private Map<Object, Object> getRandomMap(Type type) {
...
+
+        List<Object> outputList = new ArrayList<Object>() ;
```
Comment (HamzaBenyazid) [inline src/main/java/RandomGenerator.java:245]: you can use` List<T> `directly without casting

File: src/main/java/TestRandomListGenerator.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestRandomListGenerator {
+    public static void main(String[] args) throws NoSuchFieldException {
+        RandomGenerator rd = new RandomGenerator() ;
+
+        List<String> listOfString = new ArrayList<>(); // "it should be initialized like that to work"
```
Comment (HamzaBenyazid) [inline src/main/java/TestRandomListGenerator.java:8]: Not necessary. See how getRandomMap , and generateRandomObject works

**Review by HamzaBenyazid — CHANGES_REQUESTED** [review]

File: src/main/java/TestRandomListGenerator.java (hunk 4 above)
Comment (ibrahim-mestadi) [reply src/main/java/TestRandomListGenerator.java:8]: I implemented both versions by overloading

Comment (ibrahim-mestadi) [reply src/main/java/TestRandomListGenerator.java:8]: forget about test file i pushed it by mistake

**Pull request closed without merging**